torch/csrc/autograd/input_buffer.cpp

```diff
@@ -45,16 +45,44 @@ namespace torch { namespace autograd {
 
   // Switches to accumulate device
   // The device (and stream) chosen for accumulation is:
-  //  (1) If the variable is not a CUDA variable, accumulation happens on the
-  //      device of the variable.
-  //  (2) If the variable is a CUDA variable, and the producer and consumer
-  //      share its device, then:
-  //        (2a) if the producer and consumer do not share a stream,
-  //             the consumer is synced with the producer.
-  //        (2b) accumulation happens on the consumer's stream
-  //  (3) If the variable is a CUDA variable but it, the producer, and the
-  //      consumer are on multiple devices, then accumulation happens on
-  //      the default stream of the variable's device.
+  //  (1) If var is not a CUDA variable, accumulation happens on var's device.
+  //  (2) If var is a CUDA variable, and opt_producer_stream and opt_consumer_stream
```

Comment:
Simplified comment:

(1) Var is not a CUDA variable. Accumulation happens on var's device.
(2) Var is a CUDA variable and it, the consumer, and the producer share the same device:
(2a) Syncs the consumer's stream with the producer's (if different)
(2b) Accumulates on the consumer's stream
(3) Var is a CUDA variable and it shares a device with the consumer but not the producer:
(3a) Syncs the consumer's stream with the producer device's default stream
(3b) Accumulates on the consumer's stream
(4) Var is a CUDA variable and it shares a device with the producer but not the consumer:
(4a) Syncs the consumer device's default stream with the the producer's stream
(4b) Accumulates on the consumer device's default stream
(5) Var is a CUDA variable and it does not share a device with the consumer or producer. Accumulation happens on the consumer device's default stream. 

Is that correct?